ZAPD/ZString.cpp

```diff
@@ -0,0 +1,58 @@
+#include "ZString.h"
+
+#include "File.h"
+#include "ZFile.h"
+#include "StringHelper.h"
+
+REGISTER_ZFILENODE(String, ZString);
+
+ZString::ZString(ZFile* nParent) : ZResource(nParent)
+{
+}
+
+void ZString::ParseRawData()
+{
+	size_t size = 0;
+    for (size_t i = rawDataIndex; i < rawData.size(); ++i) {
+        ++size;
+        if (rawData.at(i) == '\0') {
+            break;
+        }
+    }
+
+	auto dataStart = rawData.begin() + rawDataIndex;
+	strData.assign(dataStart, dataStart + size);
+}
+
+std::string ZString::GetBodySourceCode() const
+{
+	return StringHelper::Sprintf("\t\"%s\"", strData.data());
+}
+
+std::string ZString::GetSourceOutputCode(const std::string& prefix)
+{
+	parent->AddDeclarationArray(rawDataIndex, DeclarationAlignment::None, GetRawDataSize(),
+	                            GetSourceTypeName(), name, 0, GetBodySourceCode());
+
+	return "";
+}
+
+std::string ZString::GetSourceOutputHeader(const std::string& prefix)
+{
+	return StringHelper::Sprintf("#define %s_macro \"%s\"", name.c_str(), rawData.data());
```

Comment:
Could you explain what this macro is for?

Reply:
I added this macro as a way to extract the string without actually defining it as a variable.
It could be useful if somebody wants to extract the strings used in overlay's rodata.

Reply:
Gotcha.